app/build.gradle

```diff
@@ -1,33 +1,38 @@
 apply plugin: 'com.android.application'
-
 apply plugin: 'kotlin-android'
-
 apply plugin: 'kotlin-android-extensions'
-
 apply plugin: 'kotlin-kapt'
 
 android {
-    compileSdkVersion 29
+    compileSdkVersion 30
     defaultConfig {
         applicationId "com.abedelazizshe.lightcompressor"
         minSdkVersion 21
-        targetSdkVersion 29
+        targetSdkVersion 28
```

Comment:
Why the target sdk version in downgraded?

Reply:
You don't adapt Android Q sotrage police, can't select some files from specific directory

Reply:
I was working on it actually. I pushed an update. Please resolve conflicts so I can merge. It would be great if you can test the app again where you couldn't select files before, so we can release a new version.

Thank you

Reply:
solved

Reply:
```shell
2020-08-16 09:53:49.196 5859-7223/com.abedelazizshe.lightcompressor E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-2
    Process: com.abedelazizshe.lightcompressor, PID: 5859
    java.io.FileNotFoundException: /storage/emulated/0/DCIM/Camera/VID_20200814_170121.mp4: open failed: EACCES (Permission denied)
        at libcore.io.IoBridge.open(IoBridge.java:496)
        at java.io.FileInputStream.<init>(FileInputStream.java:159)
        at com.abedelazizshe.lightcompressor.MainActivity.saveVideoFile(MainActivity.kt:212)
        at com.abedelazizshe.lightcompressor.MainActivity.access$saveVideoFile(MainActivity.kt:37)
        at com.abedelazizshe.lightcompressor.MainActivity$onActivityResult$$inlined$let$lambda$1.invokeSuspend(MainActivity.kt:95)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
     Caused by: android.system.ErrnoException: open failed: EACCES (Permission denied)
        at libcore.io.Linux.open(Native Method)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at libcore.io.BlockGuardOs.open(BlockGuardOs.java:252)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7596)
        at libcore.io.IoBridge.open(IoBridge.java:482)
        at java.io.FileInputStream.<init>(FileInputStream.java:159) 
        at com.abedelazizshe.lightcompressor.MainActivity.saveVideoFile(MainActivity.kt:212) 
        at com.abedelazizshe.lightcompressor.MainActivity.access$saveVideoFile(MainActivity.kt:37) 
        at com.abedelazizshe.lightcompressor.MainActivity$onActivityResult$$inlined$let$lambda$1.invokeSuspend(MainActivity.kt:95) 
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) 
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678) 
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665) 
```